src/cassio/table/mixins/clustered.py

```diff
@@ -23,33 +24,19 @@ def __init__(
         super().__init__(*pargs, **kwargs)
 
     def _schema_pk(self) -> List[ColumnSpecType]:
-        assert len(self.partition_id_type) == 1
-        return [
-            ("partition_id", self.partition_id_type[0]),
-        ]
+        if len(self.partition_id_type) == 1:
+            return [
+                ("partition_id", self.partition_id_type[0]),
+            ]
+        else:
+            return [
+                (f"partition_id_{pk_i}", pk_typ)
+                for pk_i, pk_typ in enumerate(self.partition_id_type)
+            ]
 
     def _schema_cc(self) -> List[ColumnSpecType]:
         return self._schema_row_id()
 
-    def _extract_where_clause_blocks(
-        self, args_dict: Any
-    ) -> Tuple[Any, List[str], Tuple[Any, ...]]:
-        """
-        If a null partition_id arrives to WHERE construction, it is silently
-        discarded from the set of conditions to create.
-        This enables e.g. ANN vector search across partitions of a clustered table.
-
-        It is the database's responsibility to raise an error if unacceptable.
-        """
-        if "partition_id" in args_dict and args_dict["partition_id"] is None:
-            cleaned_args_dict = {
-                k: v for k, v in args_dict.items() if k != "partition_id"
-            }
-        else:
-            cleaned_args_dict = args_dict
-        #
-        return super()._extract_where_clause_blocks(cleaned_args_dict)
-
```

Comment:
no longer needed in cluster mixin... completely managed in the `_normalize_kwargs()` implementation.